cfg/aws/aws_sdk_logging.go

```diff
@@ -0,0 +1,61 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package aws
+
+import (
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/influxdata/wlog"
+)
+
+var sdkLogLevel aws.LogLevelType = aws.LogOff
+
+// SetSDKLogLevel sets the global log level which will be used in all AWS SDK calls.
+// Example usage: set `debug = true` and `aws_sdk_log_level = LogDebug` in config.json.
+func SetSDKLogLevel(wLogLevel wlog.Level, sdkLogLevelString string) {
+	if wLogLevel != wlog.DEBUG {
+		sdkLogLevel = aws.LogOff
+		return
+	}
+	var temp aws.LogLevelType = aws.LogOff
+	// Hard coded strings that match actual variable names in the AWS SDK.
+	// I don't expect these names to change, or their meaning to change.
+	// This code may need updates if the SDK adds more levels (unlikely).
+	// Please note that the levels are a bit field that is OR'd together.
+	if strings.Contains(sdkLogLevelString, "LogDebug") {
+		temp |= aws.LogDebug
+	}
+	if strings.Contains(sdkLogLevelString, "LogDebugWithSigning") {
+		temp |= aws.LogDebugWithSigning
+	}
+	if strings.Contains(sdkLogLevelString, "LogDebugWithHTTPBody") {
+		temp |= aws.LogDebugWithHTTPBody
+	}
+	if strings.Contains(sdkLogLevelString, "LogDebugWithRequestErrors") {
+		temp |= aws.LogDebugWithRequestErrors
+	}
+	if strings.Contains(sdkLogLevelString, "LogDebugWithEventStreamBody") {
+		temp |= aws.LogDebugWithEventStreamBody
+	}
+
+	sdkLogLevel = temp
+}
+
+// SDKLogLevel returns the single global value so it can be used in all
+// AWS SDK calls scattered throughout the Agent.
+func SDKLogLevel() *aws.LogLevelType {
+	return &sdkLogLevel
+}
+
+// SDKLogger implements the aws.Logger interface.
+type SDKLogger struct {
+}
+
+// Log is the only method in the aws.Logger interface.
+func (SDKLogger) Log(args ...interface{}) {
+	// Always use debug logging level.
+	log.Println("D! ", args)
+}
```

Comment:
Your handling of the variadic parameters is off here.  It's treating `args` as a single argument of array type.  That's why you're getting the extra brackets in your log lines.

https://stackoverflow.com/a/56839865 shows what we need to do.

See the following scratch code that illustrates the behavior:

```
func main() {
	// 2021/12/20 11:41:09 1 2 3
	log.Println(1, 2, 3)

	// 2021/12/20 11:41:09 1 [2 3]
	helper1(2, 3)

	// 2021/12/20 11:41:09 1
	// 2021/12/20 11:41:09 2 3
	helper2(2, 3)

	// 2021/12/20 11:41:09 1 2 3
	helper3(2, 3)
}

// Like what you are doing
func helper1(args ...interface{}) {
	log.Println(1, args)
}

// I wish it could work this way, but log.Print() also adds a newline
func helper2(args ...interface{}) {
	log.Print(1)
	log.Println(args...)
}

// Ripped from StackOverflow
func helper3(args ...interface{}) {
	log.Println(append([]interface{}{1}, args...)...)
}
```

Reply:
Very cool, I will test this out now.
Thanks.

Reply:
Works like a charm:
```
2021-12-27T21:17:44Z D! DEBUG: Request monitoring/PutMetricData Details:
---[ REQUEST POST-SIGN ]-----------------------------
POST / HTTP/1.1
Host: monitoring.us-east-1.amazonaws.com
User-Agent: CWAgent/1.247349.0-42-g85a3de9b-dirty (go1.17; linux; amd64) 2021-12-27T20:56:20Z inputs:(disk mem run_as_user) outputs:(cloudwatch)
Content-Length: 697
Authorization: AWS4-HMAC-SHA256 Credential=MY_SUPER_SECRET_ACCESSKEY/20211227/us-east-1/monitoring/aws4_request, SignedHeaders=content-encoding;content-length;content-type;host;x-amz-date;x-amz-security-token, Signature=68ebd83cc48827ef78f5997eb5285c47a639409fdb568f782f39c91f1fa66445
Content-Encoding: gzip
Content-Type: application/x-www-form-urlencoded; charset=utf-8
X-Amz-Date: 20211227T211744Z
X-Amz-Security-Token: MY_SUPER_SECRET_TOKEN
Accept-Encoding: gzip


-----------------------------------------------------
2021-12-27T21:17:44Z D! DEBUG: Response monitoring/PutMetricData Details:
---[ RESPONSE ]--------------------------------------
HTTP/1.1 200 
Content-Length: 212
Content-Type: text/xml
Date: Mon, 27 Dec 2021 21:17:43 GMT
X-Amzn-Requestid: 0f185bff-51f0-4f43-9a1c-37ed4d88043c


-----------------------------------------------------
```